feat: add comprehensive Epic linking support for Jira

phuongwd opened the pull request:
# Epic Linking Support for Jira Integration

This PR adds comprehensive support for linking issues to epics in Jira, enhancing the existing integration.

## Features Added

### Core Epic Linking Functionality
- `link_issue_to_epic`: New method to intelligently link issues to epics
- `get_epic_issues`: New method to fetch all issues linked to a specific epic
- Enhanced `get_issue` to display epic relationship information

### API Endpoints and Tools
- Added `jira_link_to_epic` tool to link issues to epics
- Added `jira_get_epic_issues` tool to retrieve issues in an epic
- Improved documentation for epic-related functionality in existing tools

## Implementation Details

The implementation is robust and handles various Jira configurations:
- Tries multiple approaches to link issues (parent field, custom fields)
- Uses different JQL queries to find epic issues
- Provides clear error messages for troubleshooting

## Testing

The functionality has been manually tested with Jira Cloud.

<img width="1051" alt="image" src="https://github.com/user-attachments/assets/cb30ccd0-9713-4657-b58c-3038f5381173" />


## Related Issues
- #52 

## Credits
This feature was implemented with the assistance of Cursor AI, enhancing development efficiency and helping with robust implementation patterns.

sooperset commented:
Thanks for your contribution! From what I understand, fields like `customfield_10014`, `customfield_10000`, and `epic_link` can vary depending on the Jira instance. 
To make this more robust, I suggest adding a `get_jira_field_ids` method to the tools or including `jira_field_ids` in the resources. What do you think of this approach?

phuongwd commented:
> Thanks for your contribution! From what I understand, fields like `customfield_10014`, `customfield_10000`, and `epic_link` can vary depending on the Jira instance. To make this more robust, I suggest adding a `get_jira_field_ids` method to the tools or including `jira_field_ids` in the resources. What do you think of this approach?

@sooperset  Thanks for response.

We've implemented several fixes to address the issues with Jira integration, particularly around text formatting and issue transitions. Here's a summary of the changes:

### Key Fixes Implemented:

1. **Markdown to Jira Conversion** 
   - Added proper conversion for issue descriptions in both `create_issue` and `update_issue` methods
   - This ensures formatting like **bold** and *italic* appears correctly in Jira rather than showing literal asterisks

2. **Issue Transition Reliability**
   - Implemented a direct HTTP approach for the `jira_transition_issue` tool that bypasses problematic libraries
   - Added detailed logging to help diagnose any future issues
   - Enhanced error handling with more descriptive messages

3. **Documentation**
   - Updated the CHANGELOG to reflect all the fixes

### Testing:
The implementation now properly handles:
- Formatting in issue descriptions when creating or updating tickets
- Issue transitions with appropriate error handling
- Various Jira configurations

### Future Considerations:
- We may want to add more comprehensive test cases specifically for these formatting conversions
- Consider adding a configuration option to toggle between library-based and direct HTTP approaches for transitions
- Monitoring for any edge cases in Markdown-to-Jira conversion that might need fine-tuning

All the changes follow the project's architectural patterns and maintain backward compatibility with existing tools. The fixes are focused on improving reliability without changing the user-facing API.

Let me know if you need any clarification or have questions about the implementation details.

sooperset commented:
Thanks for the great contribution! I’ve updated the README, fixed linting, and merged it into main. Awesome work!
